Integrations/integration-Thinkst_Canary.yml

```diff
@@ -0,0 +1,558 @@
+        'id': 'ID',
+        'ip_address': 'Address',
+        'last_seen': 'LastSeen',
+        'live': 'Status',
+        'location': 'Location',
+        'name': 'Name',
+        'updated_std': 'LastUpdated',
+        'version': 'Version'
+    }
+    RELEVANT_TOKEN_ENTRIES = {
+        'canarytoken': 'CanaryToken',
+        'created_printable': 'CreatedTime',
+        'enabled': 'Status',
+        'kind': 'Kind',
+        'triggered_count': 'Triggered',
+        'doc_name': 'DocName',
+        'url': 'TokenURL'
+    }
+    session = requests.session()
+
+    '''HELPER FUNCTIONS'''
+
+
+    def get_alerts(last_fetch=None):
+        # Retrieve all unacknowledged alerts from Canary Tools
+        params = {
+            'auth_token': AUTH_TOKEN,
+        }
+
+        if last_fetch:
+            params['newer_than'] = last_fetch
+        res = session.request('GET', SERVER + 'incidents/unacknowledged', params=params, verify=not UNSECURED).json()
+        alerts = res.get('incidents')
+        return alerts
+
+
+    def create_incident(alert):
+        # Turns an alert from Canary Tools to the incident structure in Demisto
+        incident = {
+            'name': alert.get('description').get('description'),
+            'occurred': timestamp_to_datestring(1000 * (int(alert.get('description').get('created')))),
+            'rawJSON': json.dumps(alert)
+        }
+        return incident
+
+
+    '''COMMANDS'''
+
+
+    def test_module():
+        params = {
+            'auth_token': AUTH_TOKEN
+        }
+
+        res = session.request('GET', SERVER + 'ping', params=params, verify=not UNSECURED).json()
+        if res.get('result') == 'success':
+            demisto.results('ok')
+        else:
+            return_error("Test Failed")
+
+
+    def list_canaries():
+        # Retrieve all Canaries available in Canary Tools
+        params = {
+            'auth_token': AUTH_TOKEN
+        }
+
+        res = session.request('GET', SERVER + 'devices/all', params=params, verify=not UNSECURED)
+        res_json = yaml.safe_load(res.text)
+        new_devices = [
+            {new_key: device[old_key] if old_key in device else None for old_key, new_key in RELEVANT_DEVICE_ENTRIES.items()} for
+            device in res_json['devices']]
+        return res_json, new_devices
+
+
+    def list_canaries_command():
+        res_json, new_devices = list_canaries()
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res_json,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Canary Devices', new_devices,
+                                             headers=['ID', 'Name', 'Description', 'Address', 'Status', 'Location', 'Version',
+                                                      'LastSeen', 'LastUpdated']),
+            'EntryContext': {'CanaryTools.Device(val.ID && val.ID == obj.ID)': new_devices}
```

Comment:
dont initialize context in the entry, but before
use `createContext` function

Reply:
what do you mean

Reply:
this should like `'EntryContent' : context` and `context` should be initialized with that dict you have there before the entry
also make sure to use `===` operator and not `==`